04_設計/04_プラハチャレンジをDDDで実装してみる/02_APIサーバ/src/presentation/interface/route-registrar.ts

```diff
@@ -0,0 +1,22 @@
+import { IController } from "presentation/interface/controller";
+
+export interface IGetRouteRegistrar {
+  execute: (requestParams: RequestParams, controller: IController) => void;
+}
```

Comment:
[IMO]
1つのファイルから複数のインターフェースがエクスポートされている状態が個人的にあまり好きではないですね、、！
というのも、ファイル名と中身の粒度があっていないように感じるので、自分ならこういう場合は、ディレクトリを切って、その中に複数ファイルを作ってインターフェースを細かく分けるかなと思いました！
もしくは以下のように引数としてリクエスト種別を受け取るようにして、エクスポートするインターフェースを1つにするかなと思いました！
```suggestion
export interface IRouteRegistrar {
  execute: (requestType: RequestType, requestParams: RequestParams, controller: IController) => void;
}
```

Reply:
たしかに、`router-register.ts`から`IXxxRouteRegistrar`と`RequestParams`がexportされてるのは違和感があるので、ファイルを分けてexportの粒度を揃えました！
`IXxxRouteRegistrar`については、ファイルで分けても分けなくても同じかなと考え、同ファイル内にあったほうが（視覚的に）見やすいかなと思ったので同ファイル内にまとめる形にしています